lib/types.go

```diff
@@ -41,6 +42,18 @@ type Options struct {
 	Thresholds map[string][]*Threshold `json:"thresholds"`
 }
 
+type SourceData struct {
```

Comment:
So, I'm 110% on board with making a contextual file wrapper like this, it'll serve us well when we bundle up files for cloud execution as well, and I actually think we should use it *everywhere* we pass file data around.

But I don't think the runner type needs to be in it. That should be purely `run.go`'s responsibility. It doesn't make sense for a runner to get data meant for another runner. Nothing but run.go should have to concern itself with making sure the right runner gets the right kind of data.